exercises/chapter4/test/no-peeking/Solutions.purs

```diff
@@ -9,14 +9,16 @@ import Data.Maybe (Maybe(..), fromMaybe, maybe)
 import Data.Path (Path, filename, isDirectory, ls, size)
 import Data.String.Common (split)
 import Data.String.Pattern (Pattern(..))
-import Data.Tuple (Tuple(..), snd)
+import Data.Tuple (Tuple(..))
 import Test.Examples
 
 isEven :: Int -> Boolean
-isEven n = case n of
-  0 -> true
-  1 -> false
-  _ -> isEven $ n - 2
+isEven n =
+  if n == 0
+    then true
+    else if n < 0
+      then isEven (-n)
+      else not (isEven (n - 1))
```

Comment:
Deal with sign first?
```purescript
isEven n =
  if n < 0
    then isEven (-n)
    else if n == 0
      then true
      else not (isEven (n - 1))
```

Reply:
Sometimes it's nice to start with the base cases first, but I have no strong feelings either way. Applied your suggestion.